### PR TITLE
Expose ConfigurationProfiles primary and sub collection

### PR DESCRIPTION
**Pull request by bdunne**
DEPENDS_ON: https://github.com/ManageIQ/manageiq/pull/19924

### CODE REVIEW

**Comment by abellotti** [comment]
👍 just need specs. 

- [ ] spec/requests/collections_spec.rb  (for accessing as primary)
- [ ] spec/requests/providers_spec.rb   (for accessing as subcollection of providers)

File: spec/requests/providers_spec.rb [hunk 1]
```diff
@@ -1791,6 +1791,52 @@ class DummyProvider; end
...
+      it 'does not return the lans without an appropriate role' do
+        api_basic_authorize
+
+        get(api_provider_configuration_profiles_url(nil, ems))
+
+        expect(response).to have_http_status(:forbidden)
+      end
+    end
+
+    context 'GET /api/providers/:id/lans/:s_id' do
```
Comment (abellotti) [inline spec/requests/providers_spec.rb]: probably didn't mean to add this, now a dup.

Comment (bdunne) [reply spec/requests/providers_spec.rb]: Fixed

File: spec/requests/providers_spec.rb [hunk 2]
```diff
@@ -1791,6 +1791,52 @@ class DummyProvider; end
     end
   end
 
+  context 'configuration_profiles subcollection' do
+    let(:configuration_profile) { FactoryBot.create(:configuration_profile) }
+    let(:ems) { FactoryBot.create(:configuration_manager) }
+
+    context 'GET /api/providers/:id/configuration_profiles' do
+      it 'returns the lans with an appropriate role' do
```
Comment (abellotti) [inline spec/requests/providers_spec.rb]: I just realized, the it comments here and below mention lans instead of configuration profiles.

Comment (bdunne) [reply spec/requests/providers_spec.rb]: Copy/past fails again, fixed

**Comment by abellotti** [comment]
LGTM!! Thanks @bdunne for the API Enhancement. 👍 

Will merge when 🟢

**Comment by simaishi** [comment]
Jansa backport details:

```
$ git log -1
commit b0b2eb35138e16f46058fc998a449885c3617605
Author: Alberto Bellotti <abellotti@users.noreply.github.com>
Date:   Wed Mar 4 14:41:45 2020 -0500

    Merge pull request #764 from bdunne/expose_configuration_profiles

    Expose ConfigurationProfiles primary and sub collection

    (cherry picked from commit 089199ba6b81c176cf4923b5c442736ad8702f8d)
```

**Pull request merged**